pkg/doc/did/doc.go

```diff
@@ -396,6 +396,8 @@ func ParseDocument(data []byte) (*Doc, error) {
 	err := json.Unmarshal(data, &raw)
 	if err != nil {
 		return nil, fmt.Errorf("JSON marshalling of did doc bytes bytes failed: %w", err)
+	} else if raw == nil {
+		return nil, nil
```

Comment:
We cannot return `nil, nil`. It can be  `&Doc{},nil` or `nil,err`.
We have a method `validate`. I guess this check should be filtered.
there @troyronda @llorllale What do you think?

Could you please attach the panic error you got?

Reply:
Thank you for quick response. I've created issue with the panic error.

Reply:
Here is my suggestion. You need to change the `validate` function as I did below:

```
	var r *rawDoc
	if err := json.Unmarshal(data, &r); err != nil {
		return fmt.Errorf("unmarshal failed: %w", err)
	}

	if r == nil {
		return fmt.Errorf("document payload is not provided")
	}
```

Reply:
Done.

Reply:
You can do even better. I see that `validate` and `ParseDocument` functions are doing the same job. They both do `json.Unmarshal`  into `rawDoc`. It would be nice to improve optimize that by pass `rawDoc` to the  `validate` function. If `rawDoc` is provided - ignore `Unmarshal`.

Reply:
e.g 
```
func validate(data []byte, r *rawDoc) error {
	if r == nil {
		if err := json.Unmarshal(data, &r); err != nil {
			return fmt.Errorf("unmarshal failed: %w", err)
		}
	}

	if r == nil {
		return errors.New("document payload is not provided")
	}
```
and

```
func ParseDocument(data []byte) (*Doc, error) {
	var raw *rawDoc

	err := json.Unmarshal(data, &raw)
	if err != nil {
		return nil, fmt.Errorf("JSON marshalling of did doc bytes bytes failed: %w", err)
	}

	// validate did document
	if err := validate(data, raw); err != nil {
		return nil, err
	}
```

Reply:
Done